lib/Command/RecreateMasterKey.php

```diff
@@ -0,0 +1,338 @@
+	}
+
+	protected function encryptAllUsers(InputInterface $input, OutputInterface $output) {
+		/*
+		 * We are reusing the encryptAll code but not the decryptAll. The reason being
+		 * decryptAll finishes by encrypting. Which is not what we want. This will make
+		 * things out of scope for this command. We want first the entire oC FS to be
+		 * decrypt. Then re-encrypt the entire oC FS with the new master key generated.
+		 *
+		 */
+		$this->encryptAll = new EncryptAll(
+			$this->userSetup, $this->userManager, $this->rootView,
+			$this->keyManager, $this->util, $this->config,
+			$this->mailer, $this->l, $this->questionHelper,
+			$this->secureRandom);
+		$this->encryptAll->encryptAll($input, $output);
+	}
+
+	protected function decryptAllUsersFiles(ProgressBar $progress, OutputInterface $output) {
+		$userList = [];
+
+		foreach ($this->userManager->getBackends() as $backend) {
+			$limit = 500;
+			$offset = 0;
+			do {
+				$users = $backend->getUsers('', $limit, $offset);
+				foreach ($users as $user) {
+					$userList[] = $user;
+				}
+				$offset += $limit;
+			} while (count($users) >= $limit);
+		}
+
+		$userNo = 1;
+		foreach ($userList as $uid) {
+			$this->decryptUsersFiles($uid, $progress, $output);
+			$progress->advance();
+			$userNo++;
+		}
+		return true;
+	}
+
+	protected function decryptUsersFiles($uid, ProgressBar $progress, OutputInterface $output) {
+		$this->setupUserFS($uid);
+		$directories = [];
+		$directories[] = '/' . $uid . '/files';
+
+		while ($root = array_pop($directories)) {
+			$content = $this->rootView->getDirectoryContent($root);
+			foreach ($content as $file) {
+				// only decrypt files owned by the user
+				if($file->getStorage()->instanceOfStorage('OCA\Files_Sharing\SharedStorage')) {
+					continue;
+				}
+				$path = $root . '/' . $file['name'];
+				if ($this->rootView->is_dir($path)) {
+					$directories[] = $path;
+					continue;
+				} else {
+					try {
+						if ($file->isEncrypted() !== false) {
+							if ($this->decryptFile($path) === false) {
+								$progress->setMessage("decrypt files for user $uid: $path (already decrypted)");
+								$progress->advance();
+							}
+						}
+					} catch (\Exception $e) {
+						if (isset($this->failed[$uid])) {
+							$this->failed[$uid][] = $path;
+						} else {
+							$this->failed[$uid] = [$path];
+						}
+					}
+				}
+			}
+		}
+
+		if (empty($this->failed)) {
+			$this->rootView->deleteAll("$uid/files_encryption");
+		} else {
+			$output->writeln('Files for following users couldn\'t be decrypted, ');
+			$output->writeln('maybe the user is not set up in a way that supports this operation: ');
+			foreach ($this->failed as $uid => $paths) {
+				$output->writeln('    ' . $uid);
+			}
+			$output->writeln('');
+		}
+	}
+
+	protected function decryptFile($path) {
+
+		$source = $path;
+		$target = $path . '.decrypted.' . $this->getTimestamp();
+
+		try {
+			\OC\Files\Storage\Wrapper\Encryption::setDisableWriteEncryption(true);
```

Comment:
are we also using this in the occ decrypt:all command ?

Reply:
No we don't this technique of calling `setDisableWriteEncryption` at `\OC\Encryption\DecryptAll::decryptAll()`